lib/workers/repository/process/lookup/generate.ts

```diff
@@ -68,8 +69,30 @@ export async function generateUpdate(
   } else {
     update.newValue = currentValue;
   }
-  update.newMajor = versioning.getMajor(newVersion)!;
-  update.newMinor = versioning.getMinor(newVersion)!;
+
+  const newMajor = versioning.getMajor(newVersion);
+  if (is.number(newMajor)) {
+    update.newMajor = newMajor;
+  }
+
+  const newMinor = versioning.getMinor(newVersion);
+  if (is.number(newMinor)) {
+    update.newMinor = newMinor;
+  }
+
+  const newPatch = versioning.getPatch(newVersion);
+
+  // istanbul ignore next
```

Comment:
we should have a test instead to not break it again in future

Reply:
Yes, but I can't come up with test that really checks this specific behavior across datasource/lookup mechanism, and not just provides the wrong confidence. The original regression occurred exactly because it's not easy testable.

Reply:
But can't it be a simple test with two v0.0.0- versions?

Reply:
I didn't want to make it too specific, e.g. checking for particular datasource or manager. But actually maybe it's the best option in this conditions, because at least we'll know it's related to the `go` ecosystem, which isn't obvious when `updates` array length was checked for emptiness.